src/lib.rs

```diff
@@ -75,7 +75,10 @@ pub trait Field:
     const ONE: Self;
 
     /// Returns an element chosen uniformly at random using a user-provided RNG.
-    fn random(rng: impl RngCore) -> Self;
+    fn random<R: RngCore + ?Sized>(rng: &mut R) -> Self;
+
+    /// Returns an element chosen uniformly at random using a user-provided RNG.
+    fn try_from_rng<R: TryRngCore + ?Sized>(rng: &mut R) -> Result<Self, R::Error>;
```

Comment:
Same as https://github.com/zkcrypto/ff/pull/126#discussion_r1986213250, `ff-derive` needs to be updated to generate the new API.